### PR TITLE
8316387: Exclude more failing multicast tests on AIX after JDK-8315651

### PR DESCRIPTION
**Pull request by suchismith1993**
Hi all,

This pull request contains a backport of commit [c86bad51](https://github.com/openjdk/jdk21u/commit/c86bad51a88b867de3cd0b1a50bf544c2111755f) from the [openjdk/jdk21u](https://git.openjdk.org/jdk21u) repository.

The commit being backported was authored by Christoph Langer on 21 Sep 2023 and was reviewed by Matthias Baesken.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] [JDK-8316387](https://bugs.openjdk.org/browse/JDK-8316387) needs maintainer approval
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8316387](https://bugs.openjdk.org/browse/JDK-8316387): Exclude more failing multicast tests on AIX after JDK-8315651 (**Sub-task** - P4 - Approved)

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**) 🔄 Re-review required (review applies to [d3b7e56d](https://git.openjdk.org/jdk17u-dev/pull/2619/files/d3b7e56dadf18f347f549a0a681ebbdb183eb651))
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev.git pull/2619/head:pull/2619` \
`$ git checkout pull/2619`

Update a local copy of the PR: \
`$ git checkout pull/2619` \
`$ git pull https://git.openjdk.org/jdk17u-dev.git pull/2619/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2619`

View PR using the GUI difftool: \
`$ git pr show -t 2619`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/2619.diff">https://git.openjdk.org/jdk17u-dev/pull/2619.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk17u-dev/pull/2619#issuecomment-2180885851)

### CODE REVIEW

**Comment by TheRealMDoerr** [comment]
You are adding more than these 2 original changes. Especially, `macosx-all` shouldn't be part of this PR.

**Comment by suchismith1993** [comment]
> You are adding more than these 2 original changes. Especially, `macosx-all` shouldn't be part of this PR.

Yeah actually the lines got changed and it was showing as new entries. I still need to clarify. 
So for example 
java/net/MulticastSocket/Test.java                              7145658 macosx-all 
Should remain the same , as the commit is not changing it ?

**Comment by TheRealMDoerr** [comment]
A backport should only change what the original commit changes.

File: test/jdk/ProblemList.txt [hunk 1]
```diff
@@ -581,25 +581,36 @@ javax/management/remote/mandatory/connection/RMIConnector_NPETest.java 8267887 g
...
-
 java/net/ServerSocket/AcceptInheritHandle.java                  8211854 aix-ppc64
 
-
 ############################################################################
 
 # jdk_nio
 
-java/nio/channels/DatagramChannel/Unref.java                    8233437 generic-all
+
```
Comment (TheRealMDoerr) [inline test/jdk/ProblemList.txt:605]: Too many empty lines.

File: test/jdk/ProblemList.txt [hunk 2]
```diff
@@ -581,25 +581,36 @@ javax/management/remote/mandatory/connection/RMIConnector_NPETest.java 8267887 g
...
 
 java/nio/channels/AsynchronousSocketChannel/StressLoopback.java 8211851 aix-ppc64
 
+java/nio/channels/DatagramChannel/AdaptorMulticasting.java      8308807 aix-ppc64
+java/nio/channels/DatagramChannel/AfterDisconnect.java          8308807 aix-ppc64
```
Comment (TheRealMDoerr) [inline test/jdk/ProblemList.txt:610]: JDK-8316387 does not add AdaptorMulticasting.java.

File: test/jdk/ProblemList.txt [hunk 3]
```diff
@@ -581,25 +581,36 @@ javax/management/remote/mandatory/connection/RMIConnector_NPETest.java 8267887 g
...
 java/nio/channels/DatagramChannel/ManySourcesAndTargets.java    8264385 macosx-aarch64
+java/nio/channels/DatagramChannel/Unref.java                    8233437 generic-all
```
Comment (TheRealMDoerr) [inline test/jdk/ProblemList.txt:612]: JDK-8316387 does not add Unref.java.

**Review by TheRealMDoerr — COMMENTED** [review]
This version does no longer backport JDK-8316399. If this is intended, please remove it by "/issue remove JDK-8316399".

File: test/jdk/ProblemList.txt (hunk 2 above)
Comment (suchismith1993) [reply test/jdk/ProblemList.txt:610]: https://github.com/openjdk/jdk21u/commit/c86bad51a88b867de3cd0b1a50bf544c2111755f#diff-80063aaf72a19fa27f001c17fc7687086ea71b35b01f5e2b3fcd7f50d8c682fbR554

I am referring to this commit .

File: test/jdk/ProblemList.txt (hunk 3 above)
Comment (suchismith1993) [reply test/jdk/ProblemList.txt:612]: Yeah changing line number caused this.

File: test/jdk/ProblemList.txt [hunk 4]
```diff
@@ -586,19 +586,29 @@ java/net/MulticastSocket/SetLoopbackMode.java                   7122846 macosx-a
 
 java/net/MulticastSocket/Test.java                              7145658 macosx-all
 
-java/net/MulticastSocket/SetGetNetworkInterfaceTest.java        8219083 windows-all
```
Comment (TheRealMDoerr) [inline test/jdk/ProblemList.txt:589]: Please do not un-problemlist a test for windows in this PR.

**Comment by suchismith1993** [comment]
/approval request Backport not introducing new failures. Safe to backport.

**Comment by suchismith1993** [comment]
/approval request Backport not introducing new failures. Safe to backport.

File: test/jdk/ProblemList.txt [hunk 5]
```diff
@@ -587,18 +587,28 @@ java/net/MulticastSocket/SetLoopbackMode.java                   7122846 macosx-a
...
+java/net/DatagramSocket/DatagramSocketMulticasting.java         8308807 aix-ppc64
+
+java/net/MulticastSocket/B6427403.java                          8308807 aix-ppc64
+java/net/MulticastSocket/IPMulticastIF.java                     8308807 aix-ppc64
+java/net/MulticastSocket/JoinLeave.java                         8308807 aix-ppc64
+java/net/MulticastSocket/MulticastAddresses.java                8308807 aix-ppc64
+java/net/MulticastSocket/NoSetNetworkInterface.java             8308807 aix-ppc64
+java/net/MulticastSocket/Promiscuous.java                       8308807 aix-ppc64
+java/net/MulticastSocket/SetGetNetworkInterfaceTest.java        8308807 aix-ppc64
+java/net/MulticastSocket/SetOutgoingIf.java                     8308807 aix-ppc64
```
Comment (RealCLanger) [inline test/jdk/ProblemList.txt:601]: Can you please order the test exclusions like in the head commit? That is, start with java/net/DatagramSocket exclusions. Then java/net/MulticastSocket, then java/net/ServerSocket.

Please also merge the entries for java/net/MulticastSocket/SetGetNetworkInterfaceTest.java.

File: test/jdk/ProblemList.txt [hunk 6]
```diff
@@ -587,18 +587,28 @@ java/net/MulticastSocket/SetLoopbackMode.java                   7122846 macosx-a
...
 
 ############################################################################
 
 # jdk_nio
 
 java/nio/channels/DatagramChannel/Unref.java                    8233437 generic-all
-
 java/nio/channels/AsynchronousSocketChannel/StressLoopback.java 8211851 aix-ppc64
-
+java/nio/channels/DatagramChannel/AdaptorMulticasting.java      8308807 aix-ppc64
```
Comment (RealCLanger) [inline test/jdk/ProblemList.txt:610]: Please move these lines for the new exclusions to line 608, that is before java/nio/channels/DatagramChannel/Unref.java

File: test/jdk/ProblemList.txt [hunk 7]
```diff
@@ -587,18 +587,28 @@ java/net/MulticastSocket/SetLoopbackMode.java                   7122846 macosx-a
...
+java/net/MulticastSocket/SetGetNetworkInterfaceTest.java        8308807 aix-ppc64
+java/net/MulticastSocket/SetOutgoingIf.java                     8308807 aix-ppc64
 
 
 ############################################################################
 
 # jdk_nio
 
 java/nio/channels/DatagramChannel/Unref.java                    8233437 generic-all
-
```
Comment (RealCLanger) [inline test/jdk/ProblemList.txt:599]: Please keep this empy line.

**Review by RealCLanger — CHANGES_REQUESTED** [review]
I have a few change requests.

**Comment by RealCLanger** [comment]
Looks better. Still the order within java/net/MulticastSocket does not match e62419823a06ebd4499c5fadfc1f53a9bc3f71ce or what you see in head.

File: test/jdk/ProblemList.txt [hunk 8]
```diff
@@ -581,25 +581,33 @@ javax/management/remote/mandatory/connection/RMIConnector_NPETest.java 8267887 g
...
+java/net/DatagramSocket/DatagramSocketMulticasting.java         8308807 aix-ppc64
+
+java/net/MulticastSocket/B6427403.java                          8308807 aix-ppc64
+java/net/MulticastSocket/IPMulticastIF.java                     8308807 aix-ppc64
+java/net/MulticastSocket/JoinLeave.java                         8308807 aix-ppc64
+java/net/MulticastSocket/MulticastAddresses.java                8308807 aix-ppc64
 java/net/MulticastSocket/NoLoopbackPackets.java                 7122846 macosx-all
+java/net/MulticastSocket/NoSetNetworkInterface.java             8308807 aix-ppc64
+java/net/MulticastSocket/Promiscuous.java                       8308807 aix-ppc64
+java/net/MulticastSocket/SetGetNetworkInterfaceTest.java        8219083 windows-all,aix-ppc64
```
Comment (RealCLanger) [inline test/jdk/ProblemList.txt:594]: Here for SetGetNetworkInterfaceTest.java you need to add 8308807 as well. Other than that, it looks good now. :)

**Review by RealCLanger — CHANGES_REQUESTED** [review]

**Review by RealCLanger — APPROVED** [review]
LGTM now, please re-request approval.

**Comment by suchismith1993** [comment]
/approval request Backport not introducing new failures. Safe to backport.

**Pull request closed without merging**